Source/Fuse.Drawing.Surface/DotNetSurface.uno

```diff
@@ -576,9 +576,10 @@ namespace Fuse.Drawing
 			return;
 		}
 
+		/* The backing API crashes if you attempt to stroke or fill an empty path (when both dimensions are 0). But we need to still draw when one dimension is zero and the other not, since that is a veritcal/horizontal line. */
```

Comment:
How does it crash? Does it throw some exception, or does it terminate the process?

Reply:
SegFault
```
Stacktrace:

  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) System.Drawing.GDIPlus.GdipDrawPath (intptr,intptr,intptr) <0x0006b>
  at System.Drawing.Graphics.DrawPath (System.Drawing.Pen,System.Drawing.Drawing2D.GraphicsPath) <0x00192>
  at (wrapper remoting-invoke-with-check) System.Drawing.Graphics.DrawPath (System.Drawing.Pen,System.Drawing.Drawing2D.GraphicsPath) <0x0008f>
  at Fuse.Drawing.DotNetHelpers.StrokePathSolidColor (System.Drawing.Graphics,System.Drawing.Drawing2D.GraphicsPath,System.Drawing.Color,single,single,Fuse.Drawing.LineJoin,Fuse.Drawing.LineCap) <0x002f4>
  at Fuse.Drawing.DotNetSurface.StrokePath (Fuse.Drawing.SurfacePath,Fuse.Drawing.Stroke) <0x00304>
  at Fuse.Controls.Shape.Fuse.Drawing.ISurfaceDrawable.Draw (Fuse.Drawing.Surface) <0x0072c>
  at Fuse.Drawing.Surface.DrawLocal (Fuse.Drawing.ISurfaceDrawable) <0x00057>
  at Fuse.Controls.LayoutControl.ISurfaceDrawableDraw (Fuse.Drawing.Surface) <0x00818>
  at Fuse.Controls.Panel.Fuse.Drawing.ISurfaceDrawable.Draw (Fuse.Drawing.Surface) <0x00020>
  at Fuse.Drawing.Surface.DrawLocal (Fuse.Drawing.ISurfaceDrawable) <0x00057>
  at Fuse.Controls.LayoutControl.ISurfaceDrawableDraw (Fuse.Drawing.Surface) <0x00818>
  at Fuse.Controls.VectorLayer.Fuse.Drawing.ISurfaceDrawable.Draw (Fuse.Drawing.Surface) <0x00020>
  at Fuse.Drawing.Surface.DrawLocal (Fuse.Drawing.ISurfaceDrawable) <0x00057>
  at Fuse.Drawing.Surface.Draw (Fuse.DrawContext,Fuse.Elements.Element,Fuse.Drawing.ISurfaceDrawable) <0x0050b>
  at Fuse.Controls.VectorLayer.DrawWithChildren (Fuse.DrawContext) <0x00130>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.Cache.DrawHeuristically (Fuse.DrawContext) <0x0005e>
  at Fuse.Elements.Element.Composit (Fuse.DrawContext) <0x000e4>
  at Fuse.Elements.Element.Draw (Fuse.DrawContext) <0x001f3>
  at Fuse.Elements.Element.DrawNonUnderlayChildren (Fuse.DrawContext) <0x00109>
  at Fuse.Elements.Element.DrawWithChildren (Fuse.DrawContext) <0x00049>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.Cache.DrawHeuristically (Fuse.DrawContext) <0x0005e>
  at Fuse.Elements.Element.Composit (Fuse.DrawContext) <0x000e4>
  at Fuse.Elements.Element.Draw (Fuse.DrawContext) <0x001f3>
  at Fuse.Controls.Panel.Draw (Fuse.DrawContext) <0x00071>
  at Fuse.Elements.Element.DrawNonUnderlayChildren (Fuse.DrawContext) <0x00109>
  at Fuse.Elements.Element.DrawWithChildren (Fuse.DrawContext) <0x00049>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.ElementAtlas.FillFramebuffer (Fuse.DrawContext,Uno.Graphics.Framebuffer,bool,Uno.Rect) <0x00bbd>
  at Fuse.Elements.ElementAtlas.PinAndValidateFramebuffer (Fuse.DrawContext) <0x000a0>
  at Fuse.Elements.ElementBatch.Draw (Fuse.DrawContext,Uno.Float4x4,Uno.Rect) <0x00132>
  at Fuse.Elements.ElementBatcher.Draw (Fuse.DrawContext) <0x009b9>
  at Fuse.Elements.Element.DrawNonUnderlayChildren (Fuse.DrawContext) <0x00276>
  at Fuse.Elements.Element.DrawWithChildren (Fuse.DrawContext) <0x00049>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.Cache.DrawHeuristically (Fuse.DrawContext) <0x0005e>
  at Fuse.Elements.Element.Composit (Fuse.DrawContext) <0x000e4>
  at Fuse.Elements.Element.Draw (Fuse.DrawContext) <0x001c3>
  at Fuse.Controls.Panel.Draw (Fuse.DrawContext) <0x00071>
  at Fuse.Elements.Element.DrawNonUnderlayChildren (Fuse.DrawContext) <0x00109>
  at Fuse.Elements.Element.DrawWithChildren (Fuse.DrawContext) <0x00049>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.Cache.DrawHeuristically (Fuse.DrawContext) <0x0005e>
  at Fuse.Elements.Element.Composit (Fuse.DrawContext) <0x000e4>
  at Fuse.Elements.Element.Draw (Fuse.DrawContext) <0x001f3>
  at Fuse.Controls.Panel.Draw (Fuse.DrawContext) <0x00071>
  at Fuse.Elements.Element.DrawNonUnderlayChildren (Fuse.DrawContext) <0x00109>
  at Fuse.Elements.Element.DrawWithChildren (Fuse.DrawContext) <0x00049>
  at Fuse.Elements.Element.CompositEffects (Fuse.DrawContext) <0x0026e>
  at Fuse.Elements.Cache.DrawHeuristically (Fuse.DrawContext) <0x0005e>
  at Fuse.Elements.Element.Composit (Fuse.DrawContext) <0x000e4>
  at Fuse.Elements.Element.Draw (Fuse.DrawContext) <0x001f3>
  at Fuse.Controls.Panel.Draw (Fuse.DrawContext) <0x00071>
  at Fuse.Desktop.DesktopRootViewport.Draw () <0x00087>
  at Fuse.App.Draw () <0x00052>
  at Uno.Runtime.Implementation.Internal.Bootstrapper.OnDraw () <0x000d5>
  at Uno.Support.MonoMac.UnoGLView.<Initialize>m__1 (object,OpenTK.FrameEventArgs) <0x0008a>
  at OpenTK.Platform.MacOS.MonoMacGameView.OnRenderFrame (OpenTK.FrameEventArgs) <0x00032>
  at OpenTK.Platform.MacOS.MonoMacGameView.RenderScene () <0x0022d>
  at OpenTK.Platform.MacOS.MonoMacGameView.DrawRect (CoreGraphics.CGRect) <0x0006b>
  at (wrapper runtime-invoke) <Module>.runtime_invoke_void__this___CGRect (object,intptr,intptr,intptr) <0x000d0>
  at <unknown> <0xffffffff>
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend (intptr,intptr) <0x00065>
  at AppKit.NSApplication.Run () <0x0004d>
  at Uno.AppLoader.Program.Main (string[]) <0x007c1>
  at (wrapper runtime-invoke) <Module>.runtime_invoke_int_object (object,intptr,intptr,intptr) <0x00112>

Native stacktrace:


Debug info from gdb:

"monobt" command installed
(lldb) command source -s 0 '/tmp/mono-gdb-commands.NRO8cW'
Executing commands in '/tmp/mono-gdb-commands.NRO8cW'.
(lldb) process attach --pid 69677
Process 69677 stopped
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x00007fff91b9c3ee libsystem_kernel.dylib`__wait4 + 10
libsystem_kernel.dylib`__wait4:
->  0x7fff91b9c3ee <+10>: jae    0x7fff91b9c3f8            ; <+20>
    0x7fff91b9c3f0 <+12>: movq   %rax, %rdi
    0x7fff91b9c3f3 <+15>: jmp    0x7fff91b94cd4            ; cerror
    0x7fff91b9c3f8 <+20>: retq   

Executable module set to "/Users/edA-qa/src/fuselibs/Tests/ManualTests/ManualTestingApp/Surface/build/DotNet/Debug/SurfaceTestApp.app/Contents/MacOS/SurfaceTestApp".
Architecture set to: x86_64h-apple-macosx.
(lldb) thread list
Process 69677 stopped
* thread #1: tid = 0x136426, 0x00007fff91b9c3ee libsystem_kernel.dylib`__wait4 + 10, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  thread #2: tid = 0x136427, 0x00007fff91b9bbf2 libsystem_kernel.dylib`__psynch_cvwait + 10
  thread #3: tid = 0x136428, 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'tid_1103'
  thread #4: tid = 0x136429, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #5: tid = 0x13642a, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #6: tid = 0x13642b, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #7: tid = 0x136465, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #8: tid = 0x136466, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #9: tid = 0x13652d, 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
  thread #10: tid = 0x13653a, 0x00007fff91bf10c5 libsystem_malloc.dylib`szone_size + 222, name = 'tid_b23f'
  thread #11: tid = 0x13653b, 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'V8 WorkerThread'
  thread #12: tid = 0x13653c, 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'V8 WorkerThread'
  thread #13: tid = 0x13653d, 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10, name = 'V8 WorkerThread'
(lldb) thread backtrace all
* thread #1, name = 'tid_307', queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff91b9c3ee libsystem_kernel.dylib`__wait4 + 10
    frame #1: 0x00000001002b81b1 libmonosgen-2.0.dylib`mono_handle_native_sigsegv(signal=<unavailable>, ctx=<unavailable>, info=<unavailable>) at mini-exceptions.c:2348 [opt]
    frame #2: 0x000000010030dfca libmonosgen-2.0.dylib`mono_arch_handle_altstack_exception(sigctx=0x00000001000d4f48, siginfo=<unavailable>, fault_addr=<unavailable>, stack_ovf=0) at exceptions-amd64.c:808 [opt]
    frame #3: 0x000000010020c268 libmonosgen-2.0.dylib`mono_sigsegv_signal_handler(_dummy=<unavailable>, _info=<unavailable>, context=<unavailable>) at mini-runtime.c:2888 [opt]
    frame #4: 0x00007fff91c7cb3a libsystem_platform.dylib`_sigtramp + 26
    frame #5: 0x0000000124ce9f26 libcairo.2.dylib`_cairo_clip_intersect_rectangle_box(clip=0x0000000124de51c0, r=0x00007fff5fbf8298, box=0x00007fff5fbf82fc) at cairo-clip-boxes.c:179
    frame #6: 0x0000000124ce9dfd libcairo.2.dylib`_cairo_clip_intersect_box(clip=0x000000010a4bab70, box=0x00007fff5fbf82fc) at cairo-clip-boxes.c:265
    frame #7: 0x0000000124d5424c libcairo.2.dylib`_cairo_spans_compositor_stroke(_compositor=0x0000000124e10e18, extents=0x00007fff5fbf8988, path=0x0000000112877f68, style=0x00007fff5fbf8e60, ctm=0x0000000112877d20, ctm_inverse=0x0000000112877d50, tolerance=0.10000000000000001, antialias=CAIRO_ANTIALIAS_DEFAULT) at cairo-spans-compositor.c:1065
    frame #8: 0x0000000124ced40f libcairo.2.dylib`_cairo_compositor_stroke(compositor=0x0000000124e10e18, surface=0x000000010a4b95d0, op=CAIRO_OPERATOR_OVER, source=0x00007fff5fbf8e90, path=0x0000000112877f68, style=0x00007fff5fbf8e60, ctm=0x0000000112877d20, ctm_inverse=0x0000000112877d50, tolerance=0.10000000000000001, antialias=CAIRO_ANTIALIAS_DEFAULT, clip=0x000000010a4b4390) at cairo-compositor.c:157
    frame #9: 0x0000000124d095e0 libcairo.2.dylib`_cairo_image_surface_stroke(abstract_surface=0x000000010a4b95d0, op=CAIRO_OPERATOR_OVER, source=0x00007fff5fbf8e90, path=0x0000000112877f68, style=0x00007fff5fbf8e60, ctm=0x0000000112877d20, ctm_inverse=0x0000000112877d50, tolerance=0.10000000000000001, antialias=CAIRO_ANTIALIAS_DEFAULT, clip=0x000000010a4b4390) at cairo-image-surface.c:955
    frame #10: 0x0000000124d5d360 libcairo.2.dylib`_cairo_surface_stroke(surface=0x000000010a4b95d0, op=CAIRO_OPERATOR_OVER, source=0x00007fff5fbf8e90, path=0x0000000112877f68, stroke_style=0x00007fff5fbf8e60, ctm=0x0000000112877d20, ctm_inverse=0x0000000112877d50, tolerance=0.10000000000000001, antialias=CAIRO_ANTIALIAS_DEFAULT, clip=0x000000010a4b4390) at cairo-surface.c:2195
    frame #11: 0x0000000124cf8a25 libcairo.2.dylib`_cairo_gstate_stroke(gstate=0x0000000112877c30, path=0x0000000112877f68) at cairo-gstate.c:1185
    frame #12: 0x0000000124cf185e libcairo.2.dylib`_cairo_default_context_stroke(abstract_cr=0x0000000112877c00) at cairo-default-context.c:1008
    frame #13: 0x0000000124ce4aba libcairo.2.dylib`cairo_stroke(cr=0x0000000112877c00) at cairo.c:2146
    frame #14: 0x00000001155a49ad libgdiplus.dylib`stroke_graphics_with_pen(graphics=0x000000010a4b9750, pen=0x000000010a4b9d40) at graphics-cairo.c:67
    frame #15: 0x00000001155a5f70 libgdiplus.dylib`cairo_DrawPath(graphics=0x000000010a4b9750, pen=0x000000010a4b9d40, path=0x000000010a4bad60) at graphics-cairo.c:619
    frame #16: 0x00000001155a07a7 libgdiplus.dylib`GdipDrawPath(graphics=0x000000010a4b9750, pen=0x000000010a4b9d40, path=0x000000010a4bad60) at graphics.c:959
    frame #17: 0x0000000124f968fc

  thread #2
    frame #0: 0x00007fff91b9bbf2 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff91c877fa libsystem_pthread.dylib`_pthread_cond_wait + 712
    frame #2: 0x000000010042e08d libmonosgen-2.0.dylib`thread_func [inlined] mono_os_cond_wait(mutex=0x00000001005950a0) at mono-os-mutex.h:105 [opt]
    frame #3: 0x000000010042e07e libmonosgen-2.0.dylib`thread_func(thread_data=0x0000000000000000) at sgen-thread-pool.c:118 [opt]
    frame #4: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #5: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #6: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13

  thread #3, name = 'tid_1103'
    frame #0: 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x00000001003ccc9b libmonosgen-2.0.dylib`finalizer_thread [inlined] mono_os_sem_wait(flags=MONO_SEM_FLAGS_ALERTABLE) at mono-os-semaphore.h:72 [opt]
    frame #2: 0x00000001003ccc90 libmonosgen-2.0.dylib`finalizer_thread [inlined] mono_coop_sem_wait(flags=MONO_SEM_FLAGS_ALERTABLE) at mono-coop-semaphore.h:40 [opt]
    frame #3: 0x00000001003ccc85 libmonosgen-2.0.dylib`finalizer_thread(unused=<unavailable>) at gc.c:711 [opt]
    frame #4: 0x00000001003a80e1 libmonosgen-2.0.dylib`start_wrapper [inlined] start_wrapper_internal at threads.c:717 [opt]
    frame #5: 0x00000001003a7efc libmonosgen-2.0.dylib`start_wrapper(data=<unavailable>) at threads.c:764 [opt]
    frame #6: 0x000000010045a918 libmonosgen-2.0.dylib`inner_start_thread(arg=<unavailable>) at mono-threads-posix.c:92 [opt]
    frame #7: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #8: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #9: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13

  thread #4
    frame #0: 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff91c86621 libsystem_pthread.dylib`_pthread_wqthread + 1426
    frame #2: 0x00007fff91c8607d libsystem_pthread.dylib`start_wqthread + 13

  thread #5
    frame #0: 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff91c86621 libsystem_pthread.dylib`_pthread_wqthread + 1426
    frame #2: 0x00007fff91c8607d libsystem_pthread.dylib`start_wqthread + 13

  thread #6
    frame #0: 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff91c86621 libsystem_pthread.dylib`_pthread_wqthread + 1426
    frame #2: 0x00007fff91c8607d libsystem_pthread.dylib`start_wqthread + 13

  thread #7
    frame #0: 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff91c8648e libsystem_pthread.dylib`_pthread_wqthread + 1023
    frame #2: 0x00007fff91c8607d libsystem_pthread.dylib`start_wqthread + 13

  thread #8
    frame #0: 0x00007fff91c86070 libsystem_pthread.dylib`start_wqthread

  thread #9
    frame #0: 0x00007fff91b9c44e libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff91c86621 libsystem_pthread.dylib`_pthread_wqthread + 1426
    frame #2: 0x00007fff91c8607d libsystem_pthread.dylib`start_wqthread + 13

  thread #10, name = 'tid_b23f'
    frame #0: 0x00007fff91bf10c5 libsystem_malloc.dylib`szone_size + 222
    frame #1: 0x00007fff91bf0e2c libsystem_malloc.dylib`free + 87
    frame #2: 0x00000001004588ba libmonosgen-2.0.dylib`mono_thread_info_sleep [inlined] sleep_interruptable(ms=1, alerted=0x0000700004c7a694) at mono-threads.c:1157 [opt]
    frame #3: 0x00000001004586ec libmonosgen-2.0.dylib`mono_thread_info_sleep(ms=1, alerted=0x0000700004c7a694) at mono-threads.c:1185 [opt]
    frame #4: 0x00000001003a2d62 libmonosgen-2.0.dylib`ves_icall_System_Threading_Thread_Sleep_internal(ms=1) at threads.c:1181 [opt]
    frame #5: 0x00000001106e8409
    frame #6: 0x000000011063e65b
    frame #7: 0x000000011063e5b9
    frame #8: 0x000000011063dfe9
    frame #9: 0x000000010020f290 libmonosgen-2.0.dylib`mono_jit_runtime_invoke(method=<unavailable>, obj=0x00000001010741c0, params=<unavailable>, exc=<unavailable>) at mini-runtime.c:2578 [opt]
    frame #10: 0x00000001003cf262 libmonosgen-2.0.dylib`mono_runtime_invoke(method=0x00000001128c3ae8, obj=0x00000001010741c0, params=0x0000700004c7ae58, exc=0x0000000000000000) at object.c:2897 [opt]
    frame #11: 0x00000001003a8163 libmonosgen-2.0.dylib`start_wrapper [inlined] start_wrapper_internal at threads.c:723 [opt]
    frame #12: 0x00000001003a7efc libmonosgen-2.0.dylib`start_wrapper(data=<unavailable>) at threads.c:764 [opt]
    frame #13: 0x000000010045a918 libmonosgen-2.0.dylib`inner_start_thread(arg=<unavailable>) at mono-threads-posix.c:92 [opt]
    frame #14: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #15: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #16: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13

  thread #11, name = 'V8 WorkerThread'
    frame #0: 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x0000000113769af7 libV8Simple.dylib`v8::base::Semaphore::Wait() + 23
    frame #2: 0x000000011376d7d0 libV8Simple.dylib`v8::platform::TaskQueue::GetNext() + 48
    frame #3: 0x000000011376efeb libV8Simple.dylib`v8::platform::WorkerThread::Run() + 43
    frame #4: 0x000000011376b787 libV8Simple.dylib`v8::base::ThreadEntry(void*) + 87
    frame #5: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #6: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #7: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13

  thread #12, name = 'V8 WorkerThread'
    frame #0: 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x0000000113769af7 libV8Simple.dylib`v8::base::Semaphore::Wait() + 23
    frame #2: 0x000000011376d7d0 libV8Simple.dylib`v8::platform::TaskQueue::GetNext() + 48
    frame #3: 0x000000011376efeb libV8Simple.dylib`v8::platform::WorkerThread::Run() + 43
    frame #4: 0x000000011376b787 libV8Simple.dylib`v8::base::ThreadEntry(void*) + 87
    frame #5: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #6: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #7: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13

  thread #13, name = 'V8 WorkerThread'
    frame #0: 0x00007fff91b94386 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x0000000113769af7 libV8Simple.dylib`v8::base::Semaphore::Wait() + 23
    frame #2: 0x000000011376d7d0 libV8Simple.dylib`v8::platform::TaskQueue::GetNext() + 48
    frame #3: 0x000000011376efeb libV8Simple.dylib`v8::platform::WorkerThread::Run() + 43
    frame #4: 0x000000011376b787 libV8Simple.dylib`v8::base::ThreadEntry(void*) + 87
    frame #5: 0x00007fff91c8693b libsystem_pthread.dylib`_pthread_body + 180
    frame #6: 0x00007fff91c86887 libsystem_pthread.dylib`_pthread_start + 286
    frame #7: 0x00007fff91c8608d libsystem_pthread.dylib`thread_start + 13
(lldb) detach

=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

Process 69677 detached
```

Reply:
Oh boy, segfault in mono. That's nasty...

Reply:
lol

Reply:
Also nit on this line: veritcal -> vertical

Reply:
@mortoray: How about something like this instead? It moves the crash-avoidance as close to the crash-site instead...
```patch
diff --git a/Source/Fuse.Drawing.Surface/DotNetSurface.uno b/Source/Fuse.Drawing.Surface/DotNetSurface.uno
index d3d7249..248ee51 100644
--- a/Source/Fuse.Drawing.Surface/DotNetSurface.uno
+++ b/Source/Fuse.Drawing.Surface/DotNetSurface.uno
@@ -157,9 +157,6 @@ namespace Fuse.Drawing
 
 			var graphicsPath = actualPath.Path.GetGraphicsPath();
 
-			// if the path has no size, return
-			if (DotNetHelpers.IsEmptyPath(graphicsPath)) return;
-			
 			bool eoFill = actualPath.FillRule == FillRule.EvenOdd;
 			// set the path filling mode
 			DotNetHelpers.SetEOFill(graphicsPath, eoFill);
@@ -231,9 +228,6 @@ namespace Fuse.Drawing
 
 			var graphicsPath = actualPath.Path.GetGraphicsPath();
 
-			// if the path has no size, return
-			if (DotNetHelpers.IsEmptyPath(graphicsPath)) return;
-
 			bool eoFill = actualPath.FillRule == FillRule.EvenOdd;
 			// set the path filling mode
 			DotNetHelpers.SetEOFill(graphicsPath, eoFill);
@@ -576,11 +570,6 @@ namespace Fuse.Drawing
 			return;
 		}
 
-		public static extern bool IsEmptyPath(GraphicsPath path)
-		{
-			return path.GetBounds().Width == 0 || path.GetBounds().Height == 0;
-		} 
-
 		public static extern void SetEOFill(GraphicsPath path, bool eoFill)
 		{
 			path.FillMode = eoFill ? FillMode.Alternative : FillMode.Winding;
@@ -741,6 +730,8 @@ namespace Fuse.Drawing
 		{
 			var bounds = path.GetBounds();
 			bounds.Inflate(width, width);
+			if (bounds.IsEmpty)
+				return;
 
 			SolidBrush brush = new SolidBrush(color);
 			Pen pen = new Pen(brush, width);
@@ -766,6 +757,8 @@ namespace Fuse.Drawing
 		{
 			var bounds = path.GetBounds();
 			bounds.Inflate(width, width);
+			if (bounds.IsEmpty)
+				return;
 
 			var brush = new TextureBrush(image, DotNetWrapMode.Tile);
 			
@@ -798,12 +791,12 @@ namespace Fuse.Drawing
 			LineJoin lineJoin, LineCap lineCap
 		)
 		{
-			var state = graphics.Save();
 			var bounds = path.GetBounds();
-
-			// make the clipping path wider to componesate for the stroke width
 			bounds.Inflate(width, width);
+			if (bounds.IsEmpty)
+				return;
 
+			var state = graphics.Save();
 			ColorBlend blend = CreateColorBlend(lg, bounds, startX, startY, endX, endY);
 
 			var brush = new LinearGradientBrush(
@@ -848,6 +841,9 @@ namespace Fuse.Drawing
 		)
 		{
 			var bounds = path.GetBounds();
+			if (bounds.IsEmpty)
+				return;
+
 			var blend = CreateColorBlend(lg, bounds, startX, startY, endX, endY);
 
 			var startPoint = new PointF(bounds.X, bounds.Y);
@@ -893,13 +889,15 @@ namespace Fuse.Drawing
 			float width, float height
 			)
 		{
+			var bounds = path.GetBounds();
+			if (bounds.IsEmpty)
+				return;
+
 			var newImage = new Bitmap(image, (int)tileSizeX, (int)tileSizeY);
 
 			var brush = new TextureBrush(newImage, DotNetWrapMode.Tile);
 			brush.ScaleTransform(1, -1);
 			brush.TranslateTransform(originX, originY);
-			
-			var bounds = path.GetBounds();
 
 			graphics.SetClip(bounds, CombineMode.Replace);
 			graphics.FillPath(brush, path);
@@ -1271,6 +1269,7 @@ namespace Fuse.Drawing
 			public extern float X { get; set; }
 			public extern float Y { get; set; }
 			public extern void Inflate(float x,float y);
+			public extern bool IsEmpty { get; }
 		}
 
 		[DotNetType("System.Drawing.Rectangle")]
```

Reply:
@mortoray: A bit of rationale: it seems from the cairo-crash like this isn't really about the path bounds itself at all, but really about the clip-rectangle.

Reply:
In other words, I'm not 100% sure the current fix actually covers all cases that can lead to crashes (it might, but I'm not sure I would bet any money on it), and it certainly covers too many (as the need for this patch shows). But as you point out, even your case fails for a single point. Which IMO any self-respecting vector-rendering package must support.